From 0521cf2dcf3044572737df824160bde476f99a08 Mon Sep 17 00:00:00 2001
From: John Fleck <jfleck@habana.ai>
Date: Fri, 02 May 2025 22:48:34 +0300
Subject: [PATCH] [SW-227794] g3: adjust hbm temp calc in ctemp

The current HBM temperature value which is included in the
composite temperature value has a offset of 13 degrees after
the temperature reaches 80 degrees Celcius.
This can cause an oscilation when the chip temperature is
wandering above and below 80 degrees.

Modify the HBM temperature to use the following formula.

HBM_TEMP = HBM_TEMP (SoC Throttling_temp/HBM Throttling_temp)

Change-Id: Id3cface3880513fadb17e9c5a260309b3281609a
Signed-off-by: John Fleck <jfleck@habana.ai>
---

diff --git a/arch/gaudi3/ctemp.c b/arch/gaudi3/ctemp.c
index eff5d25..9d63b38 100644
--- a/arch/gaudi3/ctemp.c
+++ b/arch/gaudi3/ctemp.c
@@ -25,9 +25,10 @@
 #define CTEMP_SAMPLE_TIME_INTERVAL_MSEC	100
 #define CTEMP_UPDATE_TIME_INTERVAL_MSEC	1000
 
-#define CTEMP_HBM_SCALING_THRESHOLD	80
-#define CTEMP_HBM_SCALING_OFFSET	13
 #define CTEMP_VRM_SCALING_OFFSET	10
+
+#define SOC_THROTTLING_TEMP 118
+#define HBM_THROTTLING_TEMP 105
 
 /*
  * Some of the temp sensors can return out of range values, set a limit and
@@ -185,8 +186,8 @@
 		sample_and_check_range(config, &ctemp_info);
 
 	if (current_time - ctemp_info.update_start_time >= CTEMP_UPDATE_TIME_INTERVAL_MSEC) {
-		if (ctemp_info.max_dram_temp >= CTEMP_HBM_SCALING_THRESHOLD)
-			ctemp_info.max_dram_temp += CTEMP_HBM_SCALING_OFFSET;
+		ctemp_info.max_dram_temp =
+			(ctemp_info.max_dram_temp * SOC_THROTTLING_TEMP) / HBM_THROTTLING_TEMP;
 
 		/* On error, VRM temp will be 0, check for valid value before applying offset */
 		if (ctemp_info.max_vrm_temp >= CTEMP_VRM_SCALING_OFFSET)
